### ISSUE TITLE
Entity Unavailable

### ISSUE THREAD

**Issue by christianreiss** [al-one/hass-miio-yeelink#54]
Hey folks,

great integration; I use it for several of those Yeelink light. Working like a charm. 
The only issue I have is that when I start HA it sometimes fails 1 to all devices. The entity is then marked as "Unavailable". Restarting HA works, but this means I Have to start HA 1-3 times in a row to get all the lamps  working again.

Is there an option to get each lamp to retry connecting?

**Comment by christianreiss** [comment]
```
2022-06-09 09:55:12 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration miio_yeelink which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-06-09 09:56:28 ERROR (MainThread) [custom_components.miio_yeelink] Got exception while fetching the state for Licht Badezimmer: Unable to discover the device 10.2.1.60
```

10.2.1.60 (static ip) is ping-able from the HA vm. 

```
# nmap 10.2.1.60 -p80
Starting Nmap 7.80 ( https://nmap.org ) at 2022-06-09 09:58 CEST
Nmap scan report for 10.2.1.60
Host is up (0.070s latency).

PORT   STATE SERVICE
80/tcp open  http
MAC Address: 54:48:E6:6C:BA:75 (Beijing Xiaomi Mobile Software)
```

This is arbitrary. Re-start HA to fix this.

**Comment by I-G-1-1** [comment]
Hi, same issue here

Using _hass-miio-yeelink_ integration (installed by HACS) with two **_yeelink.light.ceiling22_** (with static IPs assigneg by the router) and HomeAssistant core 2022.6.7

The light.ceiling entity somentimes become "unavailable" and the only way to solve is to to reboot HomeAssistant.
I tried only to reload the integration in `Settings/Integration/Miio For Yeelink/ceiling01/Reload` but the light entity doesn't become available again. The only solution is reboot HA.
When unavailable in HA the lamp is still online (connected trought wifi) and operable using "Yeelight" or "Xiami Home" apps.

I think the cause of this issue is that the lamp goes offline for a short period, HA during that time don't see the ceiling lamps and correctly tag the related entity as "unavailable". Then the lamps come back online but the integration doesn't react to this event, reloading the integration doesn't solve and to regain control of the lamps HA needs to be restarted.

Basically this issue make the ceiling lamps not reliable to be operated by smart switches trought HA.

What logs can be helpful? Thanks

**Comment by al-one** [comment]
Maybe try [Xiaomi Miot Auto](https://github.com/al-one/hass-xiaomi-miot), it might work better.

**Comment by I-G-1-1** [comment]
first of all thanks for your integration and thanks for the time you spent to make this.

sorry but I don't undesrtand how to interprer your half line answer. Did you mean I don't have the skill to solve the problem or you mean I don't mind about your specific issue try this other integration?

hovewer thanks for the link and for your time

**Comment by I-G-1-1** [comment]
> Maybe try [Xiaomi Miot Auto](https://github.com/al-one/hass-xiaomi-miot), it might work better.

thanks for the help. Migrated to `Xiaomi Miot Auto` using Local connection (IP with token) and it seems to work better for now.

**Comment by I-G-1-1** [comment]
> Maybe try [Xiaomi Miot Auto](https://github.com/al-one/hass-xiaomi-miot), it might work better.

Updated today to `Xiaomi Miot Auto` version `0.6.8` and stopped working. 
I can switch on the light, enable/disable nightlight, but I cannot switch off the ligh. 
Downgraded to `0.6.7` and is working again.

**Comment by al-one** [comment]
Xiaomi Miot Auto `0.6.8` has a bug, fixed in `0.6.9`.

**Comment by I-G-1-1** [comment]
> Xiaomi Miot Auto `0.6.8` has a bug, fixed in `0.6.9`.

Thanks for the info. I can confirm that after the update to `0.6.9` all is working good again, thanks.
